syntheseus/interface/bag.py

```diff
@@ -19,8 +19,11 @@ class Bag(Collection, Generic[ElementT]):
     linear in terms of the size of the bag.
     """
 
-    def __init__(self, values: Iterable[ElementT]) -> None:
-        self._values = tuple(sorted(values))
+    def __init__(self, values: Iterable[ElementT], ordered: bool = False) -> None:
+        if ordered:
+            self._values = tuple(values)
+        else:
+            self._values = tuple(sorted(values))
```

Comment:
If we allow the elements of the `Bag` to be ordered, does that not break its interpretation as a frozen multi-set? If order matters, why not just use a `tuple`?

Reply:
I think these changes came from wanting to use the `DiskReactionDataset` class for loading data for training SMILES LMs. We may want to take a look at how much we're actually gaining from doing this.